### DISCUSSION TITLE
Have a single configuration file to increase the feedback rate of a specific robot

### DISCUSSION THREAD

**Discussion by gabrielenava** [robotology/community#578] [Ideas]
Recently in the [AMI](https://ami.iit.it/) group we tried to increase the rate at which our iCub is controlled. 

To do so, it is necessary at the moment to manually modify several configuration files in [robots-configuration](https://github.com/robotology/robots-configuration), for example:

- https://github.com/robotology/robots-configuration/tree/master/iCubGenova04/hardware/FT
- https://github.com/robotology/robots-configuration/tree/master/iCubGenova04/hardware/inertials
- https://github.com/robotology/robots-configuration/tree/master/iCubGenova04/hardware/electronics
- https://github.com/robotology/robots-configuration/tree/master/iCubGenova04/wrappers/motorControl
- https://github.com/robotology/robots-configuration/tree/master/iCubGenova04/wrappers/FT
- https://github.com/robotology/robots-configuration/tree/master/iCubGenova04/wrappers/inertials

(how many files one needs to modify depends on which information need to be streamed at an higher rate).

It would be nice (and less error-prone) to have, if possible, a single configuration file from which the user can set the rate of all sensors (e.g. FT, IMU), boards and wrappers.

cc @pattacini @DanielePucci @maggia80 @marcoaccame @traversaro

**Comment by pattacini** [comment]
cc @randaz81 @Nicogene @vtikha

**Comment by traversaro** [comment]
I think a nice way to handle this would be to extend the variable expansion capability in yarprobotinterface XML files, i.e. similar to what we did in https://github.com/robotology/yarp/pull/2819 but with the possibility of specifying arbitrary variables. In this way we could define a single variable (or few variables) in the yarprobotinteface file, and then in all the specific configuration files we refer to that variable.

**Reply by pattacini** [reply]
Nice!

**Comment by randaz81** [comment]
Actually, this functionality is already available through something like:
```
<param extern-name="externalized-param" name="my-param1"> 10 </param>
<param extern-name="externalized-param" name="my-param2"> 20 </param>
```
Then `yarprobotinterface --config test.xml --externalized-param 30` sets to 30 both my-param1 and my-param2
